tests/unit/test_req.py

```diff
@@ -636,3 +637,25 @@ def test_mismatched_versions(caplog, tmpdir):
         'Requested simplewheel==2.0, '
         'but installing version 1.0'
     )
+
+
+class TestRequirementTracker(object):
+    @pytest.fixture(autouse=True)
+    def blank_env(self):
+        with patch.dict(os.environ, clear=True):
+            yield
+
+    def test_no_side_effects_on_init(self):
+        RequirementTracker()
+        assert 'PIP_REQ_TRACKER' not in os.environ
+
+    def test_keeps_original_environ(self, tmpdir):
+        os.environ['PIP_REQ_TRACKER'] = str(tmpdir)
+        with RequirementTracker():
+            assert os.environ['PIP_REQ_TRACKER'] == str(tmpdir)
+        assert os.environ['PIP_REQ_TRACKER'] == str(tmpdir)
+
+    def test_undoes_environ_set(self):
+        with RequirementTracker():
+            assert 'PIP_REQ_TRACKER' in os.environ
+        assert 'PIP_REQ_TRACKER' not in os.environ
```

Comment:
Add a code comment noting that the variable is cleaned up.

Reply:
the test name obviates the need for a comment here imo -- let me know if you feel strongly

Reply:
I took the time to point you in the right direction for your PR, review your PR, and make the comments. I feel that all of the comments would improve the PR.

Reply:
I disagree that it's the right direction, cluttering a dead simple test with comments when the test name itself is clear seems worse.

I understand you took time to review the code and I thank you for that, but I really don't think the things you pointed out are worth another iteration.  Please push to my branch your suggestions if you feel strongly about them